tools/distpartitioning/dist_lookup.py

```diff
@@ -136,127 +141,172 @@ def get_partition_ids(self, global_nids):
         -----------
         self : instance of this class
             instance of this class, which is passed by the runtime implicitly
-        global_nids : numpy array
-            an array of global node-ids for which partition-ids are to be retrieved by
-            the distributed lookup service.
+
+        agg_global_nids : numpy array
+            an array of aggregated global node-ids for which partition-ids are
+            to be retrieved by the distributed lookup service.
 
         Returns:
         --------
         list of integers :
             list of integers, which are the partition-ids of the global-node-ids (which is the
             function argument)
         """
-
-        # Find the process where global_nid --> partition-id(owner) is stored.
-        ntype_ids, type_nids = self.id_map(global_nids)
-        ntype_ids, type_nids = ntype_ids.numpy(), type_nids.numpy()
-        assert len(ntype_ids) == len(global_nids)
-
-        # For each node-type, the per-type-node-id <-> partition-id mappings are
-        # stored as contiguous chunks by this lookup service.
-        # The no. of these mappings stored by each process, in the lookup service, are
-        # equally split among all the processes in the lookup service, deterministically.
-        typeid_counts = self.ntype_count[ntype_ids]
-        chunk_sizes = np.ceil(typeid_counts / self.world_size).astype(np.int64)
-        service_owners = np.floor_divide(type_nids, chunk_sizes).astype(
-            np.int64
+        CHUNK_SIZE = 200 * 1000 * 1000
```

Comment:
does the existing unit test cover the change? the size in test is larger than the chunk_size defined here? Just make sure chunk to multiple ones work as expected.

Reply:
Existing test case covers this case as well, but only one corner case is covered always !!!. 
To improve the test-coverage and I plan to make the chunk_sizes (they are used in 3 places in this pipeline now) to be configurable by the user itself from the command line arguments. I plan to do these changes using a separate PR. 

Once that change is done, we can run the command-line with various arguments and try testing for multiple scenarios. 

If this is acceptable, please approve the PR immediately !!!.